src/main/java/seedu/address/logic/commands/UnmatchCommand.java

```diff
@@ -0,0 +1,159 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Set;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.common.Name;
+import seedu.address.model.job.Job;
+import seedu.address.model.job.JobCompany;
+import seedu.address.model.job.JobDescription;
+import seedu.address.model.job.JobSalary;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Role;
+import seedu.address.model.skill.Skill;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Unmatches a contact from their job.
+ */
+public class UnmatchCommand extends Command {
+    public static final int CONTACT_INDEX_POS = 0;
+    public static final int JOB_INDEX_POS = 1;
+    public static final String COMMAND_WORD = "unmatch";
+    public static final String MESSAGE_USAGE =
+            COMMAND_WORD + ": Unmatch a contact to a job\nParameters: <CONTACT_INDEX> <JOB_INDEX>\nExample: "
+                    + COMMAND_WORD + " 2 1";
+
+    public static final String MESSAGE_UNMATCH_SUCCESS = "Unmatched Contact: %1$s with Job: %2$s";
+    public static final String MESSAGE_CONTACT_NOT_MATCHED = "This contact is not matched with this job!";
+    public static final String MESSAGE_CONTACT_HAS_NO_JOBS = "The contact %1$s is not associated with any job";
+
+    private final Index contactIndex;
+    private final Index jobIndex;
+
+    /**
+     * @param contactIndex Index of the contact in the filtered person list to unmatch.
+     * @param jobIndex Index of the job in the filtered job list to unmatch.
+     */
+    public UnmatchCommand(Index contactIndex, Index jobIndex) {
+        requireNonNull(contactIndex);
+        requireNonNull(jobIndex);
+
+        this.contactIndex = contactIndex;
+        this.jobIndex = jobIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Person> lastShownPeronList = model.getFilteredPersonList();
+        if (contactIndex.getZeroBased() >= lastShownPeronList.size()) {
```

Comment:
I just realised that there is a misspelling lol

Reply:
Thanks for the catch!